### PR TITLE
Add M and R aliases for cirq.measure and cirq.reset

### LINKED ISSUE #5992: cirq.M and cirq.R as aliases for measure and reset
**Is your design idea/issue related to a use case or problem? Please describe.**
I was watching someone construct a cirq Circuit, and they tried to type `cirq.M` and `cirq.R` for measure and reset, which didn't work. However, this seems like it might be a reasonable thing to support, since M and R are often in used in Circuit diagrams in similar manner as H and X?

**Describe your design idea/issue**
Make cirq.M and cirq.R aliases for `cirq.measure` and `cirq.reset`.

### PR DESCRIPTION
**Pull request by maffoo**
Fixes #5992

### CODE REVIEW

File: cirq-core/cirq/ops/common_channels.py [hunk 1]
```diff
@@ -779,6 +780,10 @@ def reset(qubit: 'cirq.Qid') -> raw_types.Operation:
     return ResetChannel(qubit.dimension).on(qubit)
 
 
+R = reset
+document(R, """Alias for cirq.reset""")
```
Comment (pavoljuhas) [inline cirq-core/cirq/ops/common_channels.py:784]: This overwrites the reset docstring [here](https://github.com/quantumlib/Cirq/blob/24a0d2e60caa25bad135ea8b737beb308e4fad57/cirq-core/cirq/_doc.py#L44).

Perhaps we should just add a note about the `R` alias to the reset docstring?

**Review by pavoljuhas — CHANGES_REQUESTED** [review]
We need to keep the actual docstrings for measure and reset.

File: cirq-core/cirq/ops/common_channels.py (hunk 1 above)
Comment (pavoljuhas) [reply cirq-core/cirq/ops/common_channels.py:784]: If we want references to cirq.M in the generated docs, we might need to create a wrapper function so it is a different object.

**Comment by maffoo** [comment]
Thanks, @pavoljuhas. I fixed the docstrings and added notes there about the new aliases.

**Pull request merged**